tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideBadFlagTestCases
+     * @param $headerValue
+     * @param $expectedErrors
+     */
+    public function testBadFlags($headerValue, $expectedErrors)
+    {
+        $BadCSP = HeaderFactory::build('Content-Security-Policy', $headerValue);
+
+        $Errors = CSPBadFlags::validate($BadCSP);
+
+        if ([] === $expectedErrors)
+        {
+            $this->assertCount(0, $Errors);
+        }
+        else
+        {
+            $this->assertEquals($expectedErrors, $Errors);
```

Comment:
In testing it appears to check count first, then manually compare elements with equivalent of `==` (i.e. check items are instances of same class, then iterate though all properties of the class and check their equality).

E.g. changing the expected `Error` warning level results in:
<img width="822" alt="screen shot 2017-07-15 at 19 42 31" src="https://user-images.githubusercontent.com/3288888/28241893-0106239c-6996-11e7-8e6b-e61148e6a8bb.png">